cellmlmanip/parser.py

```diff
@@ -310,41 +312,49 @@ def _add_variables(self, component_element):
 
         return variable_lookup_symbol
 
-    def _add_maths(self, component_element, variable_to_symbol):
+    def _add_maths(self, maths_to_add, connected_variable_mapping):
         """
+        Add maths for all elements sorted in maths_to_add.
+
         <model> <component> <math> </component> </model>
 
-        :param component_element: an etree.Element
-        :param variable_to_symbol: a ``Dict[str, sympy.Dummy]``
+        :param maths_to_add: a list of (component_element, variable_to_symbol) tuples
+                             where ``component_element`` is an ``etree.Element`
+                             and ``variable_to_symbol`` is a ``Dict[str, sympy.Dummy]``
+
+        :param connected_variable_mapping: a ``Dict[str, sympy.Dummy]`` mapping a connected variable to its source.
```

Comment:
Wondering about `maths_to_add` as a variable name: I was expecting it to contains maths (e.g. maths elements or sympy equations), but it contains component elements. Looking through the code I think you're right that this is the most efficient way to do it, but maybe the name should be changed?

Reply:
you're right of course better like this?